tools/matinfo/src/main.cpp

```diff
@@ -866,15 +878,169 @@ static void transpileSpirv(const std::vector<uint32_t>& spirv) {
     std::cout << glslCompiler.compile();
 }
 
-static void disassembleSpirv(const std::vector<uint32_t>& spirv) {
+// Consumes SPIRV binary and produces an ordered map from "line number" to "GLSL string" where
+// the line number is determined by line directives, and the GLSL string is one or more lines of
+// transpiled GLSL-ES.
+static std::map<int, std::string> transpileSpirvToLines(const std::vector<uint32_t>& spirv) {
+    using namespace spirv_cross;
+
+    CompilerGLSL::Options emitOptions;
+    emitOptions.es = true;
+    emitOptions.vulkan_semantics = true;
+    emitOptions.emit_line_directives = true;
+
+    CompilerGLSL glslCompiler(move(spirv));
+    glslCompiler.set_common_options(emitOptions);
+    std::string transpiled = glslCompiler.compile();
+
+    std::map<int, std::string> result;
+    const std::regex lineDirectivePattern("\\#line ([0-9]+)");
+
+    std::istringstream ss(glslCompiler.compile());
+    std::string glslCodeline;
+    int currentLineNumber = -1;
+    while (std::getline(ss, glslCodeline, '\n')) {
+        std::smatch matchResult;
+        if (std::regex_search(glslCodeline, matchResult, lineDirectivePattern)) {
+            currentLineNumber = stoi(matchResult[1].str());
+        } else {
+            result[currentLineNumber] += glslCodeline + "\n";
+        }
+    }
+
+    return result;
+}
+
+static void analyzeSpirv(const std::vector<uint32_t>& spirv, const char* disassembly) {
+    using namespace std;
+
+    const map<int, string> glsl = transpileSpirvToLines(spirv);
+    const regex globalDecoratorPattern("OpDecorate (\\%[A-Za-z_0-9]+) RelaxedPrecision");
+    const regex typeDefinitionPattern("(\\%[A-Za-z_0-9]+) = OpType[A-Z][a-z]+");
+    const regex memberDecoratorPattern("OpMemberDecorate (\\%[A-Za-z_0-9]+) ([0-9]+) RelaxedPrecision");
+    const regex structDefinitionPattern("(\\%[A-Za-z_0-9]+) = OpTypeStruct");
```

Comment:
I don't see where you're using `structDefinitionPattern` anywhere?

Reply:
Removed.